submission/persistence/src/main/kotlin/ac/uk/ebi/biostd/persistence/util/SubmissionFilterSpecification.kt

```diff
@@ -0,0 +1,57 @@
+package ac.uk.ebi.biostd.persistence.util
+
+import ac.uk.ebi.biostd.persistence.model.Submission
+import ac.uk.ebi.biostd.persistence.model.User
+import org.springframework.data.jpa.domain.Specification
+
+class SubmissionFilterSpecification(userId: Long, filter: SubmissionFilter) {
+    var specification: Specification<Submission>
+
+    init {
+        specification = Specification.where(
+            withUser(userId)).and(withVersionGreaterThan(0))
+        filter.accNo?.let { specification = specification.and(withAccession(it)) }
+        filter.rTimeFrom?.let { specification = specification.and(withFrom(it)) }
+        filter.rTimeTo?.let { specification = specification.and(withTo(it)) }
+        filter.keywords?.let {
+            if (filter.keywords.isNotBlank())
+                specification = specification.and(withTitleLike(it))
+        }
+    }
+
+    private fun withVersionGreaterThan(version: Int): Specification<Submission> {
+        return Specification { root, query, cb ->
+            cb.greaterThan<Int>(root.get<Int>("version"), version)
+        }
+    }
```

Comment:
some comments about this. 

1. They can be inliners which looks way better.
```
    private fun withVersionGreaterThan(version: Int): Specification<Submission> =
        Specification { root, _, cb -> cb.greaterThan<Int>(root.get<Int>("version"), version) }
```

2.  Not used parameter can be replaced in kotlin with `_` to reduce cognitive load. so it will become
```
   private fun withVersionGreaterThan(version: Int): Specification<Submission> =
        Specification { root, _, cb -> cb.greaterThan<Int>(root.get<Int>("version"), version) }
```

3. We have a specific class for storing well know submission properties that we can use here to avoid typo in constants and facilitate refactoring if we require in the feature. Check `ebi.ac.uk.model.constants.SubFields`